### DISCUSSION TITLE
Allow generic type inference from a method.

### DISCUSSION THREAD

**Discussion by BreadTh** [dotnet/csharplang#5595] [Language Ideas] [answered]
Allow generic type inference from a method.
From the following code it appears that generic type inference doesn’t attempt to infer from the types in a method signature, as it does from a delegate.
At the same time, assigning a method to a variable will correctly infer the delegate signature from the method.

```csharp
void Execute<T>(Action<T> func) { }
void Method(object obj) { }

var action = Method;     //var is Action<object>?

Execute(Method);         //doesn't work, CS0411
Execute<object>(Method); //works
Execute(action);         //works
```

There may be technical obstacles that prevents this from becoming a reality, but here’s how I imagine it: If the parser can automatically cast a method to a delegate, and it can automatically infer the generic type from a delegate, why shouldn’t it be able to do both in one step?

**Comment by jnm2** [comment, marked as answer]
I believe this might be a duplicate of https://github.com/dotnet/csharplang/discussions/129.

**Reply by BreadTh** [reply]
That seems to be the case. I'll close this thread and cross my fingers for progress on #129.
